// Copyright 2021 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// The Pigweed device RPC proto definition.
// The source of truth of this proto can be found at https://github.com/project-chip/connectedhomeip/tree/master/examples/common/pigweed/protos
syntax = "proto3";

import "third_party/py/gazoo_device/protos/common.proto";

package chip.rpc;

// type lengths defined in chip spec 8.2.3.1
message DeviceInfo {
  uint32 vendor_id = 1;
  uint32 product_id = 2;
  uint32 software_version = 3;
  string serial_number = 4;
}

service Device {
  // To factory reset the device.
  rpc FactoryReset(pw.protobuf.Empty) returns (pw.protobuf.Empty){}

  // To reboot the device.
  rpc Reboot(pw.protobuf.Empty) returns (pw.protobuf.Empty){}

  // To trigger the device OTA.
  rpc TriggerOta(pw.protobuf.Empty) returns (pw.protobuf.Empty){}

  // To get the device information.
  rpc GetDeviceInfo(pw.protobuf.Empty) returns (pw.protobuf.Empty){}
}
